core/src/main/java/org/apache/carbondata/core/datamap/dev/DataMap.java

```diff
@@ -42,6 +44,15 @@
    */
   List<Blocklet> prune(FilterResolverIntf filterExp);
 
+  /**
+   * Prune the datamap with blockletId. It returns the list of
+   * blocklets where these filters can exist.
+   *
+   * @param filterExp
+   * @param blockletId
+   * @return
+   */
+  List<Blocklet> pruneBlockletFromBlockId(FilterResolverIntf filterExp, int blockletId);
```

Comment:
what is blockletId? I don't think this method is required in datamap

Reply:
BlockletId is the output of Min Max DataMap and the same is passed to BlockletDataMap in order to form the complete blocklet. 
Instead of declaring the method pruneBlockletFromBlockId in DataMap, the same can be made a local function to blockletId.